### ISSUE TITLE
Editor: Gallery layout broken on Gutenberg posts 

### ISSUE THREAD

**Issue by loremattei** [wordpress-mobile/WordPress-iOS#10355]
From a TestFlight user feedback:

- Start a post on web, using GB as the editor;
- Add a gallery with 4 images;
- Edit the post on WPiOS;
- Save it

The layout of the post is broken, with the images no longer visible.

**Comment by hypest** [comment]
> Edit the post on WPiOS;
Can we expand a bit on what exactly gets edited? Thanks!

**Comment by diegoreymendez** [comment]
Nothing that I am aware of.

**Comment by etoledom** [comment]
I'm trying to reproduce the issue without success.
What version of WPiOS and Gutenberg are you using? @loremattei @koke 

I'm on WPiOS 11.1, Gutenberg web v4.1.1 (.com atomic site)

I did notice two interesting details, that might (or might not) be related.

**The first one**, just by creating and publishing the gallery, I see this behavior on web (without intervention from the mobile app):

![gutenberg_web_gallery](https://user-images.githubusercontent.com/9772967/47672850-c9f7c800-db91-11e8-84c4-b4ff2e819c56.gif)

https://etoledomatomicsite01.blog/2018/10/29/gallery/

----

**The second one** is more interesting, and related to Aztec. When I edit adding a paragraph after the gallery, it adds it as another gallery block:

![edir_gallery](https://user-images.githubusercontent.com/9772967/47674090-4049f980-db95-11e8-8e20-4ca9c00317ee.png)

And back to the web side, it's not recognized:
![screen shot 2018-10-29 at 4 09 09 pm](https://user-images.githubusercontent.com/9772967/47674116-4f30ac00-db95-11e8-93e9-241f7d0bb220.png)

**Comment by koke** [comment]
Make sure at least one of the images has a caption set

**Comment by etoledom** [comment]
Thank you @koke !
That triggered some more weird behavior that is definitely breaking Gutenberg web. But still I don't see the images disappearing.

I also tested with other themes, thinking that it might be related to themes too, but no luck there either.

This is what I see:
Just after publishing from Gutenberg web
![screen shot 2018-10-29 at 5 05 26 pm](https://user-images.githubusercontent.com/9772967/47677424-d3872d00-db9d-11e8-81b5-9ef0d7f97bb9.png)

After saving from WPiOS:
![screen shot 2018-10-29 at 5 06 17 pm](https://user-images.githubusercontent.com/9772967/47677462-e7329380-db9d-11e8-8b05-7062940be760.png)

Also tested adding caption to all the images:
![screen shot 2018-10-29 at 5 08 29 pm](https://user-images.githubusercontent.com/9772967/47677477-f6b1dc80-db9d-11e8-9368-3e64d2019792.png)

For now I will investigate this one. If it's not exactly the same issue, at least it seems to be related. (There's something wrong saving galleries).

**Comment by koke** [comment]
That matches what I saw, I didn't have any missing images

**Comment by loremattei** [comment]
This is very close to the behaviour I got while trying to reproduce the issue, with the difference that, for some reasons, the gallery size in my test didn't grew, so I started with 4 images and then I had only the first two, because the other two slots were used by the captions. 
I didn't look at the code, so I don't know what happened exactly. I'll try to redo it again, since I messed up the test post while looking at other reports. 

Anyway, this is the original message: 
> If you use Gutenberg as the Editor and you use the Wordpress own Galleries, WP4iOS destroys them. E.g. You get insted of the 4 pictures, you need the place for 8, behause the Titel/Caption standa besite the pictures. 

so I think that the reported issue is the one @etoledom is reproducing.
